tests/components/roku/test_config_flow.py

```diff
@@ -0,0 +1,247 @@
+        side_effect=RokuException,
+    ) as mock_validate_input:
+        result = await async_configure_flow(hass, result["flow_id"], {CONF_HOST: HOST},)
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+    await hass.async_block_till_done()
+    assert len(mock_validate_input.mock_calls) == 1
+
+
+async def test_form_cannot_connect_request(hass: HomeAssistantType) -> None:
+    """Test we handle cannot connect request error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.roku.config_flow.validate_input",
+        side_effect=RequestException,
+    ) as mock_validate_input:
+        result = await async_configure_flow(hass, result["flow_id"], {CONF_HOST: HOST},)
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+    await hass.async_block_till_done()
+    assert len(mock_validate_input.mock_calls) == 1
+
+
+async def test_form_cannot_connect_socket(hass: HomeAssistantType) -> None:
+    """Test we handle cannot connect socket error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.roku.config_flow.validate_input",
+        side_effect=SocketGIAError,
+    ) as mock_validate_input:
+        result = await async_configure_flow(hass, result["flow_id"], {CONF_HOST: HOST},)
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+    await hass.async_block_till_done()
+    assert len(mock_validate_input.mock_calls) == 1
+
+
+async def test_form_unknown_error(hass: HomeAssistantType) -> None:
+    """Test we handle unknown error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.roku.config_flow.validate_input",
+        side_effect=Exception,
+    ) as mock_validate_input:
+        result = await async_configure_flow(hass, result["flow_id"], {CONF_HOST: HOST},)
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "unknown"
+
+    await hass.async_block_till_done()
+    assert len(mock_validate_input.mock_calls) == 1
+
+
+async def test_import(hass: HomeAssistantType) -> None:
+    """Test the import step."""
+    with patch(
+        "homeassistant.components.roku.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.roku.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result = await async_init_flow(
+            hass, context={CONF_SOURCE: SOURCE_IMPORT}, data={CONF_HOST: HOST}
+        )
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == HOST
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5184723)